services/language_translation/v2.js

```diff
@@ -52,327 +78,322 @@ module.exports = function (RED) {
-              shape: 'ring',
-              text: 'call to translation service failed'
-            });
-            node.error(err, msg);
-          } else {
-            msg.payload = model.status;
-            node.send(msg);
-            node.status({});
-          }
-        }
-      );
-    }
-
-    this.doDelete = function(msg, trainid) {
-      var trainid = config.trainid;
-
-      if (!trainid) {
-        node.warn('Missing ID, please enter one');
+      if (!action) {
+        node.warn('Missing action, please select one');
         node.send(msg);
         return;
       }
 
-      node.status({
-        fill: 'blue',
-        shape: 'dot',
-        text: 'deleting'
-      });
+      // We have credentials, and know the mode. Further required fields checks
+      // are specific to the requested action.
+      // The required fields are checked, before the relevant function is invoked.
+      switch (action) {
+      case 'translate':
+        var domain = msg.domain || config.domain;
 
-      language_translation.deleteModel({ model_id: trainid},
-        function(err) {
-          node.status({});
-          if (err) { 
-            node.status({
-              fill: 'red',
-              shape: 'ring',
-              text: 'could not delete'
-            });
-            node.error(err, msg);
-          } else {
-            msg.payload = "model deleted";
-            node.send(msg);
-            node.status({});
-          }
+        if (!domain) {
+          node.warn('Missing translation domain, message not translated');
+          node.send(msg);
+          return;
         }
-      );      
-    }
+        
+        var srclang = msg.srclang || config.srclang;
+        
+        if (!srclang) {
+          node.warn('Missing source language, message not translated');
+          node.send(msg);
+          return;
+        }
+        
+        var destlang = msg.destlang || config.destlang;
+        
+        if (!destlang) {
+          node.warn('Missing target language, message not translated');
+          node.send(msg);
+          return;
+        }
+        
+        var model_id = '';
+        
+        model_id = srclang + '-' + destlang;
+        if (domain !== 'news') {
+          model_id += ('-' + domain);
+        }
+        doTranslate(msg, model_id);
+        break;
+      
+      case 'custom':
+        var custom = msg.custom || config.custom;
+      
+        if (!custom) {
+          node.warn('Missing customised model, message not translated');
+          node.send(msg);
+          return;            
+        } 
+        doTranslate(msg, custom);
+        break; 
+      
+      case 'train':
+        var basemodel = msg.basemodel || config.basemodel;
+      
+        if (!basemodel) {
+          node.warn('Base Model needs must be set for train mode');
+          node.send(msg);            
+        }
+        var filetype = msg.filetype || config.filetype;
+        if (!filetype) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 10 space characters but found 11.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2613390925/issues/source?bid=3225115&fileBranchId=3311899#l363)